src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -640,6 +649,15 @@ private SafeCurlSlistHandle SetRequestHeaders(SafeCurlHandle handle, HttpRequest
                     rawHandle = Interop.libcurl.curl_slist_append(rawHandle, header);
                     retVal.SetHandle(rawHandle);
                 }
+
+                // Since libcurl always adds a Transfer-Encoding header, we need to explicitly block
+                // it if caller specifically does not want to set the header
+                if (request.Headers.TransferEncodingChunked.HasValue && !request.Headers.TransferEncodingChunked.Value)
+                {
+                    rawHandle = Interop.libcurl.curl_slist_append(rawHandle, NoTransferEncoding);
```

Comment:
What happens if NoTransferEncoding was one of the headers in allHeader set previously?  Is that possible?  If it is possible, is it a problem we need to prevent, or is it ok to let the header be set twice?